### PR TITLE
Small changes to memory doc

### PR DESCRIPTION
**Pull request by chick**
- Fixed typo "except" => "accept"
- Use `Counter` explicitly in ROM section example.

### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [NA] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### API Impact

None

#### Backend Code Generation Impact

Not applicable

#### Desired Merge Strategy

Squash

#### Release Notes

Fixes to memory docs, fixed typo and improved ROM example

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (3.2.x, 3.3.x, 3.4.x, 3.5.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

File: docs/src/explanations/memories.md [hunk 1]
```diff
@@ -24,8 +24,12 @@ def sin(t: Double): Double = t // What should this be?
 For example, users can create a small ROM initialized to 1, 2, 4, 8 and loop through all values using a counter as an address generator as follows:
 
 ```scala mdoc:compile-only
-val m = VecInit(1.U, 2.U, 4.U, 8.U)
-val r = m(counter(m.length.U))
```
Comment (seldridge) [inline docs/src/explanations/memories.md:28]: I finally see what's going on here... Can you also remove `scala mdoc:invisible` block above. That was what was defining `counter`, but just not showing it. Not ideal.

**Review by seldridge — APPROVED** [review]

File: docs/src/explanations/memories.md (hunk 1 above)
Comment (chick) [reply docs/src/explanations/memories.md:28]: Done

File: src/main/scala/chisel3/playground.sc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
+val Pi = math.Pi
+val x = math.sin(Pi / 4)
```
Comment (jackkoenig) [inline src/main/scala/chisel3/playground.sc:3]: Was this a debugging playground?

File: docs/src/explanations/memories.md [hunk 3]
```diff
@@ -11,31 +11,28 @@ Chisel provides facilities for creating both read only and read/write memories.
...
+  import chisel3._
+  import chisel3.util.Counter
+  val m = VecInit(1.U, 2.U, 4.U, 8.U)
+  val c = Counter(m.length)
+  c.inc()
+  val r = m(c.value)
 ```
```
Comment (jackkoenig) [inline docs/src/explanations/memories.md:25]: Can you untable this? Nothing else is tabbed

File: src/main/scala/chisel3/playground.sc (hunk 2 above)
Comment (chick) [reply src/main/scala/chisel3/playground.sc]: Yes, dang, deleted.

File: docs/src/explanations/memories.md (hunk 3 above)
Comment (chick) [reply docs/src/explanations/memories.md:25]: Removed tabs

**Comment by chick** [comment]
Heading towards highest commits per lines changed ever!

**Pull request merged**